crates/provider/src/utils.rs

```diff
@@ -23,75 +17,16 @@ fn estimate_priority_fee(rewards: Vec<Vec<U256>>) -> U256 {
     if rewards.is_empty() {
         return U256::ZERO;
     }
-    if rewards.len() == 1 {
-        return rewards[0];
-    }
-    // Sort the rewards as we will eventually take the median.
-    rewards.sort();
-
-    // A copy of the same vector is created for convenience to calculate percentage change
-    // between subsequent fee values.
-    let mut rewards_copy = rewards.clone();
-    rewards_copy.rotate_left(1);
-
-    let mut percentage_change: Vec<I256> = rewards
-        .iter()
-        .zip(rewards_copy.iter())
-        .map(|(a, b)| {
-            let a = I256::try_from(*a).expect("priority fee overflow");
-            let b = I256::try_from(*b).expect("priority fee overflow");
-            ((b - a) * I256::try_from(100).expect("Unexpected overflow")) / a
-        })
-        .collect();
-    percentage_change.pop();
-
-    // Fetch the max of the percentage change, and that element's index.
-    let max_change = percentage_change.iter().max().unwrap();
-    let max_change_index = percentage_change.iter().position(|&c| c == *max_change).unwrap();
-
-    // If we encountered a big change in fees at a certain position, then consider only
-    // the values >= it.
-    let values = if *max_change
-        >= I256::from_raw(U256::from(EIP1559_FEE_ESTIMATION_THRESHOLD_MAX_CHANGE))
-        && (max_change_index >= (rewards.len() / 2))
-    {
-        rewards[max_change_index..].to_vec()
-    } else {
-        rewards
-    };
 
+    rewards.sort();
     // Return the median.
-    values[values.len() / 2]
-}
-
-fn base_fee_surged(base_fee_per_gas: U256) -> U256 {
-    if base_fee_per_gas <= U256::from(40_000_000_000u64) {
-        base_fee_per_gas * U256::from(2)
-    } else if base_fee_per_gas <= U256::from(100_000_000_000u64) {
-        base_fee_per_gas * U256::from(16) / U256::from(10)
-    } else if base_fee_per_gas <= U256::from(200_000_000_000u64) {
-        base_fee_per_gas * U256::from(14) / U256::from(10)
-    } else {
-        base_fee_per_gas * U256::from(12) / U256::from(10)
-    }
+    rewards[rewards.len() / 2]
```

Comment:
```suggestion
    let n = rewards.len();
    if n % 2 == 0 {
        (rewards[n / 2] + rewards[n / 2 + 1]) / 2
    } else {
        rewards[(n + 1) / 2]
    }
```

Reply:
wouldn't this panic when n == 2?

Reply:
sure, just flagging this is not the real median

Reply:
22e3f0df96b23ac1efde66fbe8ebe61cd50d1bd3